services/accounts.service.js

```diff
@@ -0,0 +1,84 @@
+import Account from "../models/accounts.model.js";
+import Currency from "../models/currencies.model.js";
+
+const accountsService = {
+  addAccountIdParam: (req) => {
+    req.accountId = req.params.accountId;
+  },
+
+  insertAccount: async (req, res) => {
+    const checkCurrency = await Currency.findById(req.body.currency);
+    if (!checkCurrency)
+      res.status(400).json({ message: "Provided currency does not exist." });
```

Comment:
I dont like this. You response set status and message a lot times. It's better to move this logic to middlware (exception handler) and just throw exceptions here. Ideally service shouldnt know what request/response is (objects res and req are not passed to service). Controllers and services are different layers. Controller knows about service but service shouldnt know about controller